### ISSUE TITLE
Add getters for camera parameters

### ISSUE THREAD

**Issue by Nokse22** [f3d-app/f3d#1419]
**Is your feature request related to a problem? Please describe.**
When using libf3d and an external window you will  need to use yaw, azimuth, elevation to move but to get them in degree you would need to calculate them from the camera position.

**Describe the solution you'd like**
It could be useful to have getters for these parameters. I know they are not actually parameters, but ways to manipulate the camera, but a getter for yaw/elevation that returns the angle of the camera to the model can be useful, also distance from the model etc.

**Suggested implementation**
In `camera.h` and `camera.cxx` , add `camera::getYaw`, `camera::getAzimuth`,  `camera::getElevation`. Implement them to be computed on the fly.

**Comment by mwestphal** [comment]
> but a getter for yaw/elevation that returns the angle of the camera to the model can be useful

I suppose they could be computed for easier usage of the class indeed.

> distance from the model etc.

This one is way more complex, unless you mean distance to the focal point.

**Comment by Nokse22** [comment]
> This one is way more complex, unless you mean distance to the focal point.

Yes, from the focal point.

**Comment by Schwarzemann** [comment]
I'd like to work on this.

**Comment by mwestphal** [comment]
Do you need help moving forward @Schwarzemann ?

**Comment by Schwarzemann** [comment]
> Do you need help moving forward @Schwarzemann ?

Ah no I am fine thanks. I accidentally deleted all of my commits so far so I am restoring them.

**Comment by mwestphal** [comment]
unassigning for now.

**Comment by dvaldivia1** [comment]
Hi this is Diego from the discord server!

**Comment by AbbasReads** [comment]
Hi! I'm new to open source. I'd like to work on this. Please help me get started.

**Comment by mwestphal** [comment]
no feedback from @dvaldivia1 , assiging to @AbbasReads then :)

**Comment by mwestphal** [comment]
Of course @AbbasReads , please join the discord as well to discuss: https://discord.f3d.app

**Comment by mwestphal** [comment]
Hi @AbbasReads , need any help moving forward ?

**Comment by AbbasReads** [comment]
Working on it.

**Comment by mwestphal** [comment]
@AbbasReads is not available currently, lets unassign for now.

**Comment by Dtsitos** [comment]
Hello there,  can I work on this issue :)?

**Comment by mwestphal** [comment]
Indeed! Please note this was almost finished, so you need to finish an ongoing feature :)

**Comment by mwestphal** [comment]
no feedback, unassigning

**Comment by code2766** [comment]
Hi! I'm a C++ developer looking to make my first contribution. I'd love to take on this issue to add the camera parameter getters. Could you please assign it to me?

**Comment by mwestphal** [comment]
Sure, please note this was started and not finished here: https://github.com/f3d-app/f3d/pull/2767

**Comment by prathambalehosurr** [comment]
Hi! I'd like to work on this issue as my first contribution to F3D. Is it still available? Any guidance on where in the codebase I should start looking would be appreciated.

**Comment by mwestphal** [comment]
Hi @prathambalehosurr I'm afraid its already assigned, please pick another issue

**Comment by prathambalehosurr** [comment]
> Hi [@prathambalehosurr](https://github.com/prathambalehosurr) I'm afraid its already assigned, please pick another issue

Thanks for letting me know! I'll look through the remaining good first issues and pick another one. Looking forward to contributing.

**Comment by mwestphal** [comment]
Need any help moving forward ? @code2766

**Comment by mwestphal** [comment]
Need any help moving forward ? @code2766

**Comment by mwestphal** [comment]
no feedback, unassigning
